### ISSUE TITLE
CSV Output

### ISSUE THREAD

**Issue by Arrowstar** [Ren0k/Project-Atmospheric-Drag#1]
Hello!

This tool is great but the JSON output can be hard to understand.  Could you please create a CSV file output that just has a table of the Mach number and drag coefficients, if possible?  Or at least the inputs needed to compute said coefficients?  This would be easier for the average user to understand and work with.

Thank you!

**Comment by Ren0k** [comment]
## JSON File
The JSON output file is structured to be imported directly by kOS, to create a lexicon. 
See this section on [serialization.](https://ksp-kos.github.io/KOS/commands/serialization.html)
I did not design the output file to be used by external sources, hence there is no option to save it as a CSV or other filetype. 
I do think it is a good idea, I will look into this.
  
## A simple solution
It should actually be fairly easy to construct a csv file from the json.
You can import the JSON file with kOS and this will create a lexicon object.
A simple script should be able to convert the lexicon into other filetypes.

**Comment by Arrowstar** [comment]
Great, I'm glad to hear that the CSV solution is not that challenging to implement.  The reason that JSON import doesn't work for me well is that I'm using your data outside of kOS in an external application.  I have the ability to read JSON files but what comes out of that is really not pleasant to use.  Thank you for your help!

Btw, would it be possible to augment the data that is computed with not only drag coefficient vs Mach number, but also total angle of attack as well?  This would be a 2D matrix of data with each row and column corresponding to a Mach number or angle of attack.

**Comment by Ren0k** [comment]
> Great, I'm glad to hear that the CSV solution is not that challenging to implement. The reason that JSON import doesn't work for me well is that I'm using your data outside of kOS in an external application. I have the ability to read JSON files but what comes out of that is really not pleasant to use. Thank you for your help!
> 
> Btw, would it be possible to augment the data that is computed with not only drag coefficient vs Mach number, but also total angle of attack as well? This would be a 2D matrix of data with each row and column corresponding to a Mach number or angle of attack.

The way a drag profile is structured currently, is that it corresponds to a fixed angle of attack. 
The original idea of a drag profile was to provide drag information for ballistic atmospheric flight, where typically the vessel is alligned with the surface velocity vector. This would allow for determination of impact position.
As an optional feature I added a custom angle of attack option for specific situations where the vessel would have an offset angle to the velocity vector.
You can create multiple drag profiles, each one for a different angle of attack, and combine the data later. 
  
To allow for the creation of a matrix style drag profile that includes a range of angle of attack values, I have to rewrite the drag profile creation function. In addition the functions that use drag profiles will have to be adjusted, so this is a bigger task.

Having said that, I do like the idea of creating files that can be used externally in for example data analysis.
When I return to the project in the near future, the first major task is to rewrite the project entirely and this allows me to make significant changes and additions to the core elements.
Thanks for the feedback and suggestions, I'll add them to the list!

**Comment by Arrowstar** [comment]
Great, thank you!

`As an optional feature I added a custom angle of attack option for specific situations where the vessel would have an offset angle to the velocity vector.
You can create multiple drag profiles, each one for a different angle of attack, and combine the data later.`
Is this functionality already in the code, and if so, how do I use it?

**Comment by Ren0k** [comment]
> 

Yes, you can find all the extra features under 'configure' in the part-analysis menu.

**Comment by Arrowstar** [comment]
Great, thanks!

I look forward to the updates we talked about whenever you have a chance to implement them. :)
